client/src/components/auth/ResetModal.js

```diff
@@ -0,0 +1,141 @@
+import React, { Component } from 'react';
+import {
+  Button,
+  Modal,
+  ModalHeader,
+  ModalBody,
+  Form,
+  FormGroup,
+  Label,
+  Input,
+  NavLink,
+  Alert
+} from 'reactstrap';
+//import 'bootstrap/dist/css/bootstrap.min.css';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { resetPassword } from '../../actions/resetActions';
+import { clearErrors } from '../../actions/errorActions';
+import { clearSuccess } from '../../actions/authActions';
+
+class ResetModal extends Component{
+
+    state = {
+        modal: false,
+        newPassword: null,
+        verifyPassword: null,
+        msg: null,
+    };
+
+    static propTypes = {
+        isAuthenticated: PropTypes.bool.isRequired,
+        error: PropTypes.object.isRequired,
+        resetPassword: PropTypes.func.isRequired,
+        clearErrors: PropTypes.func.isRequired,
+        success: PropTypes.bool.isRequired
+    };
+
+    componentDidUpdate(prevProps) {
+        const { error, isAuthenticated, success } = this.props;
+        if (error !== prevProps.error) {
+          // Check for register error
+          if (error.id === 'PASSWORD_RESET_FAIL') {
+            this.setState({ msg: error.msg.msg });
+          } else {
+            this.setState({ msg: null });
+            
+          }
+        }
+
+        if(success !== prevProps.success){
+            this.setState({msg: 'Password Updated Successfully!!!'});
+            //this.props.clearSuccess();
+        }
+        // If authenticated, close modal
+        // If authenticated, close modal
+        
+    }
+
+    toggle = () => {
+        this.props.clearErrors();
+        this.setState({
+            modal: !this.state.modal,
+        });
+        this.props.clearSuccess();
+        //console.log(this.state.modal);
+    };
+
+    onChange = (e) => {
+        this.setState({ [e.target.name]: e.target.value });
+    };
+
+    onSubmit = e => {
+        this.props.clearErrors();
+        e.preventDefault();
+        const { newPassword, verifyPassword } = this.state;
+        const Password = {
+            newPassword,
+            verifyPassword
+        };
+        this.props.resetPassword(Password);
```

Comment:
added changes in resetActions file, it checks whether passwords match before sending request to server.

So changes are made in 'resetActions' instead of resetModal.

Reply:
What you can do instead is put the state as success as true and also save the success msg to the state via the reducers and then display it in the FirstPage.jsx using alert and after a timeout clear the success and the msg.

Reply:
But isn't it working fine now? as in the success, the message is shown correctly
I have made the changes in the reducer only. The primary reason why I didn't make changes to the fistPasge.jsx was there are other models as well, for example, we can add a similar success message for Register Modal as well, and controlling the success messages using a 
single success variable would lead to chaos.
Therefore these success messages should be added to their respective models only and not to first page.

Do you agree?

Reply:
Having a more common approach to all the success messages is much better and the current success msg is kinda in a wierd place also so it makes sense to add it in the main page. What do you say about this @Prajwal7842 ??

Reply:
Okay I have gone through the pr. Firstly I think that the current format goes with the message that is displayed when resetting the password. But according to me, there should be a common state variable success and msg that should be reused again and again(as suggested by Rohit). Also this would not be chaotic but actually simple because as the project grows, it will not be feasible to manage different variables for different types success message. Also the reset password would not be like a tab(my opinion) in the navbar but rather would be something in a profile page maybe(not sure about this). Rerouting to the first page and then displaying the message would be better in this scenario.
Besides these implementation details, the functionality is working very good and I really likes the code structure,  I think after some changes this can be merged.

Reply:
Video of new success msg
pls have a look
[https://www.youtube.com/watch?v=rHJiPprHOgo](url)

Reply:
You didn't have to upload a youtube video, you could have attached a zip file or converted into a gif and attached it. I guess the success msg looks fine but I asked you to remove the second success msg!!

Reply:
About the second success message he(@MegaMind77-coder ) told me that we would do it before.

Reply:
Removed old alert

Reply:
Pls review again

Reply:
Looks good.